src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -298,71 +369,117 @@ class Definitions {
-
-  lazy val PairClass                    = ctx.requiredClass("dotty.Pair")
-  lazy val PartialFunctionClass         = ctx.requiredClass("scala.PartialFunction")
-  lazy val AbstractPartialFunctionClass = ctx.requiredClass("scala.runtime.AbstractPartialFunction")
-  lazy val SymbolClass                  = ctx.requiredClass("scala.Symbol")
-  lazy val ClassClass                   = ctx.requiredClass("java.lang.Class")
-  lazy val DynamicClass                 = ctx.requiredClass("scala.Dynamic")
-  lazy val OptionClass                  = ctx.requiredClass("scala.Option")
-  lazy val BoxedNumberClass             = ctx.requiredClass("java.lang.Number")
-  lazy val ThrowableClass               = ctx.requiredClass("java.lang.Throwable")
-  lazy val ClassCastExceptionClass      = ctx.requiredClass("java.lang.ClassCastException")
-  lazy val JavaSerializableClass        = ctx.requiredClass("java.lang.Serializable")
-  lazy val ComparableClass              = ctx.requiredClass("java.lang.Comparable")
-  lazy val ProductClass                 = ctx.requiredClass("scala.Product")
-    lazy val Product_canEqual = ProductClass.requiredMethod(nme.canEqual_)
-    lazy val Product_productArity = ProductClass.requiredMethod(nme.productArity)
-    lazy val Product_productPrefix = ProductClass.requiredMethod(nme.productPrefix)
-  lazy val LanguageModuleClass          = ctx.requiredModule("dotty.language").moduleClass.asClass
-  lazy val NonLocalReturnControlClass   = ctx.requiredClass("scala.runtime.NonLocalReturnControl")
+  lazy val NullPointerExceptionClass = ctx.requiredClass("java.lang.NullPointerException")
+  lazy val ClassClass                = ctx.requiredClass("java.lang.Class")
+  lazy val BoxedNumberClass          = ctx.requiredClass("java.lang.Number")
+  lazy val ThrowableClass            = ctx.requiredClass("java.lang.Throwable")
+  lazy val ClassCastExceptionClass   = ctx.requiredClass("java.lang.ClassCastException")
+  lazy val JavaSerializableClass     = ctx.requiredClass("java.lang.Serializable")
+  lazy val ComparableClass           = ctx.requiredClass("java.lang.Comparable")
+
+  // in scalac modified to have Any as parent
+
+  lazy val SerializableType: TypeRef       = ctx.requiredClassRef("scala.Serializable")
+  def SerializableClass(implicit ctx: Context) = SerializableType.symbol.asClass
+  lazy val StringBuilderType: TypeRef      = ctx.requiredClassRef("scala.collection.mutable.StringBuilder")
+  def StringBuilderClass(implicit ctx: Context) = StringBuilderType.symbol.asClass
+  lazy val MatchErrorType: TypeRef         = ctx.requiredClassRef("scala.MatchError")
+  def MatchErrorClass(implicit ctx: Context) = MatchErrorType.symbol.asClass
+
+  lazy val StringAddType: TypeRef          = ctx.requiredClassRef("scala.runtime.StringAdd")
+  def StringAddClass(implicit ctx: Context) = StringAddType.symbol.asClass
+
+    lazy val StringAdd_plusR = StringAddClass.requiredMethodRef(nme.raw.PLUS)
```

Comment:
It's systematic R instead of Ref. But it's probably better to make them private anyway.